### PR TITLE
Add godoc to AutoEvent and correct frequency type

### LINKED ISSUE #24: Add godoc to AutoEvent and correct the type of frequency
The type of frequency ought to be uint32 rather than int32
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/models/autoevent.go#L20

Also, we had better add some document into autoevent.go

### PR DESCRIPTION
**Pull request by cloudxxx8**
fix #24
Signed-off-by: Cloud Tsai <cloudxxx8@gmail.com>

### CODE REVIEW

**Comment by cloudxxx8** [comment]
@iain-anderson please help review and correct my wording.  thanks

File: models/autoevent.go [hunk 1]
```diff
@@ -16,10 +16,21 @@ package models
 
 import "encoding/json"
 
+/*
+ * This file is the model for the AutoEvent object in EdgeX
```
Comment (tsconn23) [inline models/autoevent.go]: Maybe instead "`AutoEvent supports auto-generated events sourced from a device service`"

File: models/autoevent.go [hunk 2]
```diff
@@ -16,10 +16,21 @@ package models
...
+ *
+ * AutoEvent struct
+ */
 type AutoEvent struct {
-	Frequency int32  `json:"frequency,omitempty"`
-	OnChange  bool   `json:"onChange,omitempty"`
-	Resource  string `json:"resource,omitempty"`
+	// Frequency indicates that how often the specific resource need to be polled.
```
Comment (tsconn23) [inline models/autoevent.go]: "`Frequency indicates how often the specific resource needs to be polled.`"

File: models/autoevent.go [hunk 3]
```diff
@@ -16,10 +16,21 @@ package models
...
+	// It represents the number of events per day.
+	// For example, 86400 would be once per second, 24 would be onece per hour etc.
```
Comment (tsconn23) [inline models/autoevent.go]: "once" misspelled as "onece"

File: models/autoevent.go [hunk 4]
```diff
@@ -16,10 +16,21 @@ package models
...
+	Frequency uint32 `json:"frequency,omitempty"`
+	// OnChange indicates whether generate the events only if the reading value is different from the previous one.
```
Comment (tsconn23) [inline models/autoevent.go]: "`OnChange indicates whether the device service will generate an event only if the reading value is different from the previous one.`"

**Comment by cloudxxx8** [comment]
@tsconn23 thanks for the suggestion, I've corrected the according to your comment

**Pull request merged**